lib/connection-manager.js

```diff
@@ -81,15 +86,15 @@ const checkHealth = (config) => {
         utils.log('streaming resumed after healthcheck call')
       }
 
-      return setTimeout(() => {
+      config.healthcheckTimer = setTimeout(() => {
         return checkHealth(config)
       }, config.HEALTHCHECK_INTERVAL) // Every 30 Minutes
     })
   }
 }
 
 // Initiate Connection
-const connectLogServer = (config) => {
+function connectLogServer(config) {
```

Comment:
it would be helpful if this (and `checkHealth`) accepted a callback so we didn't need the sleep in tests, and just in general.

out of scope here probably just something that stood out.

Reply:
True.  I can addd a `TODO`.  The code does so much with different platforms that I'm at a loss with how to add more tests without doing the planned `major` release--but that will require solving some issues with our test approach on different OS's.